src/details/ArborX_DetailsBoundingVolumeHierarchyImpl.hpp

```diff
@@ -420,6 +439,8 @@ void BoundingVolumeHierarchyImpl<DeviceType>::queryDispatch(
                     } );
             } );
         Kokkos::fence();
+
+        Kokkos::Profiling::popRegion();
     }
     // do not copy if by some miracle each query exactly yielded as many results
     // as the buffer size
```

Comment:
Should we add a region for the copy if the buffer optimization succeeded?  (This is not a rhetorical question)

Reply:
We'll have a counter there for parallel_for region anyway, so no.

Reply:
Virtually all have counters for parallel regions so I am not convinced by your argument